pkg/controller/sparkapplication/sparkapp_metrics.go

```diff
@@ -196,6 +218,18 @@ func (sm *sparkAppMetrics) exportMetrics(oldApp, newApp *v1beta2.SparkApplicatio
 		}
 	}
 
+	// In the event that state transitions happened too quickly and the spark app skipped the RUNNING state, the job
+	// start latency should still be captured.
+	// Note: There is an edge case that a Submitted state can go directly to a Failing state if the driver pod is
+	//       deleted. This is very unlikely if not being done intentionally, so we choose not to handle it.
+	if newState != oldState {
+		if (newState == v1beta2.FailingState || newState == v1beta2.SucceedingState) && oldState != v1beta2.RunningState {
```

Comment:
The condition after `&&` probably should be `oldState == v1beta2.SubmittedState` as I don't think it makes sense to update the metric if the oldState is `PendingRerun` for example.

Reply:
Sure, that sounds good to me